core/src/main/java/org/jruby/RubySymbol.java

```diff
@@ -150,15 +153,57 @@ public final String asJavaString() {
         return symbol;
     }
 
+    /**
+     * Return an id string (e.g. raw ISO-8859_1 charset String) for use with our method tables etc.
+     */
+    public String idString() {
+        return symbol;
+    }
+
     @Override
     public final String toString() {
-        return symbol;
```

Comment:
this could use an explanation on why not simply `return symbol;` ...for mere mortals :)

Reply:
Ah yeah I will add an explanatory comment.  Largely toString is doing it's best to make a j.l.String which displays properly but is not perfect since toString() should not be used internally for anything but debugging.  Going full RubyString + decode seemed unnecesary here.  Perhaps I should also add asJavaString() or using str() is more appropriate if you want to work with a string version of a RubySymbol?

Reply:
overriding `asJavaString` explicitly (from IRubyObject) sounds good ... for the full decoding cycle.
unless its already doing so

Reply:
I also just found a few consumers of toString and changed them to idString or asJavaString depending on whether their context was used in an identifier or non-identifier way (for RubySymbol).  I will likely be trying to audit more general toString() usage as we seem to be a bit inconsistent.

Reply:
@kares oh I should point out asJavaString cannot be decoded version of the method since there are still over 100 users using it to lookup methods.  We have this pattern:

```java
    public static IRubyObject get_inner_class(final ThreadContext context,
        final RubyModule self, final IRubyObject name) { // const_missing delegate
        final String constName = name.asJavaString();
```

All our internal tables for identifiers are keyed off of 8859_1.   I think we may need to make a slightly more restrictive but helpful method to replace this pattern and ween off of asJavaString (idString on RubySymbol is more correct but only on RubySymbol -- probably need something on RubyString as well).  This though need not be resolved immediately.  Just letting you know why asJavaString cannot decode today.

As an aside it is a little weird IRubyObject has asJavaString since this means I could pass a fixnum in here and it would end up as "5" or something.  From Ruby side of things I think people would be surprised that would work even if there was a "5" method.